crates/proto/src/event.rs

```diff
@@ -0,0 +1,73 @@
+use anyhow;
+use std::collections::HashMap;
+use tendermint::abci;
+use crate::{Name, Message};
+
+
+pub trait ProtoEvent: Message + Name + Sized {
+
+    fn into_event(&self) -> abci::Event {
+
+        let kind  = Self::NAME;
+
+        // NOTE: Clarify if the serde derivation is pbjson/prost aware. Or, more specifically, if Penumbra generated serialization code for protobuf defs are more coherent than the 
+        //       3-4 different ways the cosmo-sdk chooses to serializes and deserialize its protobuf data in the reference code.
+        // TODO: Would it be preferred to return an empty(?) or error coded abci::Event?
+        let event_json = serde_json::to_vec(&self.encode_to_vec()).expect("ProtoEvent constrained values should be JSON serializeable.");
```

Comment:
this is clarifying. at some point I realized that I didn't know if there was a cost/benefit to relying on those traits being transitively resolved because of `Message`.